map.go

```diff
@@ -44,6 +48,13 @@ type MapSpec struct {
 	BTF *btf.Map
 }
 
+// Specify numa node during map creation
+func (ms *MapSpec) WithNumaNode(node uint32) *MapSpec {
```

Comment:
I'm a bit torn about this. It adds a function for a very seldomly used feature. Plus, the function signature is very confusing: it suggests that the original MapSpec isn't modified, which isn't true.

I'd prefer it if you removed the function, and the only thing we keep is the NumaNode field + documentation.

Reply:
If we remove the function, we must re-expose the `unix.BPF_F_NUMA_NODE` flag, which is in an internal package, or apply the flag automatically when `NumaNode` is set.